### PR TITLE
NAS Benchmark

### PR DESCRIPTION
**Pull request by ultmaster**
- [x] Add NAS-Bench-201
- [x] Add NAS-Bench-101
- [x] NDS
- [x] Test end-to-end installation script
- [x] Test API
- [x] Documentation

### CODE REVIEW

File: examples/nas/benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* To avoid storage and legal issues, we do not provide any generated databases. Users have to generate the database on your own. All generation scripts can be found in `examples/nas/benchmark`. To ease the burden of installing multiple dependencies, we strongly recommend users to use docker to run the generation scripts.
+* Please prepare a folder to household all the benchmark databases. By default, it can be found at `${HOME}/.nni/nasbenchmark`. You can place it anywhere you like, and specify it in `NASBENCHMARK_DIR` before importing NNI.
+* Please install `peewee` by `pip install peewee`, which NNI uses to connect to database.
+
+## NAS-Bench-101
+
+### Preparation
+
+```bash
+docker run -e NNI_VERSION=${NNI_VERSION} -v ${HOME}/.nni/nasbenchmark:/outputs .:/root tensorflow/tensorflow:1.15.2-py3 /bin/bash /root/nasbench101.sh
```
Comment (QuanluZhang) [inline examples/nas/benchmark/README.md:16]: is there docker image in this command?

File: examples/nas/benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+Please replace `${NNI_VERSION}` with any NNI version, for example, v1.6 or master.
+
+It takes about 70 minutes to dump the records and build index. Output size is about 1.9GB.
```
Comment (QuanluZhang) [inline examples/nas/benchmark/README.md:21]: how can I store the generated dataset out of docker image?

File: examples/nas/benchmark/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+### API Documentation
```
Comment (QuanluZhang) [inline examples/nas/benchmark/README.md:23]: please complete this part

File: examples/nas/benchmark/nasbench101.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
+set -e
+mkdir -p /outputs /tmp
```
Comment (QuanluZhang) [inline examples/nas/benchmark/nasbench101.sh:2]: it is not friendly to use `\outputs`

File: src/sdk/pynni/nni/nas/benchmark/nasbench101/graph_util.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+def adjancency_matrix_from_architecture(architecture, vertices):
+    matrix = np.zeros((vertices, vertices), dtype=np.bool)
+    for i in range(1, vertices):
+        for k in architecture['input{}'.format(i)]:
+            matrix[k, i] = 1
+    return matrix
+
+
+def nasbench_format_to_architecture_repr(adjacency_matrix, labeling):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/benchmark/nasbench101/graph_util.py:20]: please add docstring

File: src/sdk/pynni/nni/nas/benchmark/nasbench201/__init__.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+from .model import Nb201ComputedStats, Nb201IntermediateStats, Nb201RunConfig
+from .shortcuts import query_nb201_computed_stats
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/benchmark/nasbench201/__init__.py:2]: what is the meaning of `shortcuts`?

Comment (ultmaster) [reply src/sdk/pynni/nni/nas/benchmark/nasbench201/__init__.py:2]: I followed the practice from some other projects. Do you have a better name in your mind?

File: examples/nas/benchmark/nasbench101.sh (hunk 4 above)
Comment (ultmaster) [reply examples/nas/benchmark/nasbench101.sh:2]: It's in docker image.

File: examples/nas/benchmark/README.md (hunk 2 above)
Comment (ultmaster) [reply examples/nas/benchmark/README.md:21]: You mount a folder into `/outputs` in docker, and everything in that folder will be persisted.

File: examples/nas/benchmark/README.md (hunk 1 above)
Comment (ultmaster) [reply examples/nas/benchmark/README.md]: Docker image is `tensorflow/tensorflow:1.15.2-py3`

File: docs/en_US/nas.rst [hunk 7]
```diff
@@ -23,4 +23,5 @@ For details, please refer to the following tutorials:
     One-shot NAS <NAS/one_shot_nas>
     Customize a NAS Algorithm <NAS/Advanced>
     NAS Visualization <NAS/Visualization>
+    NAS Benchmark <NAS/Benchmark>
```
Comment (QuanluZhang) [inline docs/en_US/nas.rst:26]: Benchmark -> Benchmarks

File: examples/nas/benchmark/nasbench101.sh (hunk 4 above)
Comment (QuanluZhang) [reply examples/nas/benchmark/nasbench101.sh:2]: fine

File: src/sdk/pynni/nni/nas/benchmark/nasbench101/shortcuts.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import functools
+
+from peewee import fn
+from playhouse.shortcuts import model_to_dict
+from .model import Nb101ComputedStats, Nb101IntermediateStats, Nb101RunConfig
+from .graph_util import hash_module, infer_num_vertices
+
+
+def query_nb101_computed_stats(arch, num_epochs, isomorphism=True, reduction=None):
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/benchmark/nasbench101/shortcuts.py:9]: is "computed_stats" a common terminology?

File: src/sdk/pynni/nni/nas/benchmark/nasbench201/__init__.py (hunk 6 above)
Comment (QuanluZhang) [reply src/sdk/pynni/nni/nas/benchmark/nasbench201/__init__.py:3]: query.py

File: src/sdk/pynni/nni/nas/benchmark/nasbench101/model.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    Run config for NAS-Bench-101.
+
+    Attributes
+    ----------
+    arch : dict
+        A dict with keys ``op1``, ``op2``, ... and ``input1``, ``input2``, ... Vertices are
+        enumerate from 0. Since node 0 is input node, it is skipped in this dict. Each ``op``
+        is one of :const:`nni.nas.benchmark.nasbench101.CONV3X3_BN_RELU`,
+        :const:`nni.nas.benchmark.nasbench101.CONV1X1_BN_RELU`, and :const:`nni.nas.benchmark.nasbench101.MAXPOOL3X3`.
+        Each ``input`` is a list of previous nodes. For example ``input5`` can be ``[0, 1, 3]``.
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/nas/benchmark/nasbench101/model.py]: could you give a complete example of `arch`? it is fine to have only several nodes

**Pull request merged**